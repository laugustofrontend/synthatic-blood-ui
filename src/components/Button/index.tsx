const Button = () => (
  <div>
    <h1 className="text-2xl text-red-600 font-bold underline">Button</h1>
  </div>
)

export default Button
