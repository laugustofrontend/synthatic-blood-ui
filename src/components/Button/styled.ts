import tw from 'tailwind-styled-components'

import { ButtonProps } from '.'

export const Button = tw.button<ButtonProps>`
  bg-gray-300
  text-black-500
  rounded-md
  px-4
  py-2
`
