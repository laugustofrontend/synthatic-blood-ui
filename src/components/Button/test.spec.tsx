import { describe, expect, test as it } from 'vitest'
import { render, screen } from '@testing-library/react'

import Button from './index'

describe('<Button />', () => {
  it('should render the <Button />', () => {
    const { container } = render(<Button />)

    expect(screen.getByRole('heading', { name: /Button/i })).toBeDefined()

    expect(container.firstChild).toMatchSnapshot()
  })
})
