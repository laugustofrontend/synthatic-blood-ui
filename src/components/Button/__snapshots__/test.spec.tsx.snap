// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should render the <Button /> 1`] = `
<div>
  <h1
    class="text-2xl text-red-600 font-bold underline"
  >
    Button
  </h1>
</div>
`;
